### PR TITLE
switch to vite and vitest

### PR DESCRIPTION
**Pull request by ToppleTheNun**
### Description

vite is much faster than create-react-app
and create-react-app is also stagnant.

a lot of the changes are a result of an auto-import cleanup that
was done to help with the vitest migration.

### Testing

You need to pull it and test it yourself. :)

### CODE REVIEW

File: Dockerfile.dev [hunk 1]
```diff
@@ -6,6 +6,6 @@ RUN npm i -g pnpm
 
 COPY . /usr/src/wowanalyzer/
 
-EXPOSE 3000
```
Comment (emallson) [inline Dockerfile.dev:7]: would prob be good to keep the development port the same if possible to easily/reliably configure. we have enough stuff changing, don't need to add more.

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      babel: {
+        plugins: ['@emotion/babel-plugin', 'macros'],
+      },
+    }),
+    tsconfigPaths(),
+    {
+      name: 'vite-plugin-wowanalyzer-index-html-inject-ga',
+      transformIndexHtml: (html) =>
+        process.env.VITE_ENABLE_GA === 'true'
+          ? html.replace('</head>', GOOGLE_ANALYTICS_SCRIPT)
```
Comment (emallson) [inline vite.config.ts:63]: a little spicy, but clever

**Review by emallson — COMMENTED** [review]
i have only skimmed this, excited to try it out

File: Dockerfile.dev (hunk 1 above)
Comment (ToppleTheNun) [reply Dockerfile.dev:7]: I will repoint everything to 3000

**Pull request closed without merging**